Sources/spm-test-finder/main.swift

```diff
@@ -0,0 +1,48 @@
+#if os(OSX)
+
+import XCTest
+import Foundation
+
+enum Error: ErrorProtocol {
+    case invalidUsage
+    case unableToLoadBundle(String)
+}
+
+do {
+    guard Process.arguments.count == 2 else {
+        throw Error.invalidUsage
+    }
+    var bundlePath = Process.arguments[1]
+    if !(bundlePath as NSString).isAbsolutePath {
+        bundlePath = NSFileManager.default().currentDirectoryPath + "/" + bundlePath
+    }
+    bundlePath = (bundlePath as NSString).standardizingPath
+
+    guard let bundle = NSBundle(path: bundlePath) where bundle.load() else {
```

Comment:
As mentioned on the [JIRA ticket](https://bugs.swift.org/browse/SR-1541?focusedCommentId=14671&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-14671), an issue can arise here if the test bundle code writes to stdout during loading, or during test discovery. To make this tool's output more robust, stdout should be redirected to `/dev/null` during these steps. I left an example of this in C in JIRA.